pkg/apis/core/types.go

```diff
@@ -4643,6 +4643,18 @@ type Preconditions struct {
 	UID *types.UID
 }
 
+// LogStreamType represents the desired log stream type.
+type LogStreamType string
+
+const (
+	// LogStreamTypeStdout is the stream type for stdout.
+	LogStreamTypeStdout LogStreamType = "stdout"
```

Comment:
I don't see mention of this in the KEP or discussion, but this is a constant in our public API and should follow constant rules.  For instance, `all` should be `All`.  While `stdout` and `stderr` are terms of art, they are generally not directly referenced via name except in documentation.  The POSIX spec *does* reference them in their lower case form (https://pubs.opengroup.org/onlinepubs/9699919799/functions/stdin.html) but does not define how to use them.  I would slightly lean towards `stderr` over `Stderr` or `StdErr` or `STDERR`, but will get input from other reviewers.

So go ahead and change `all` to `All` unless there is previous discussion justifying the non-standard constant, and I may come back and suggest a change to the other constant values depending on other opinions.